rmw_fastrtps_cpp/src/rmw_client.cpp

```diff
@@ -261,7 +264,8 @@ rmw_destroy_client(rmw_node_t * node, rmw_client_t * client)
     }
     delete info;
   }
-  delete (client);
+  rmw_free(const_cast<char *>(client->service_name));
+  rmw_free(client);
```

Comment:
This should use the symmetric function to `rmw_client_allocate` which is `rmw_client_free`.

Similar for the service below.